### PR TITLE
Examples: auth0

### PR DESCRIPTION
**Pull request by krystian50**
Auth0 example code

### CODE REVIEW

**Comment by vasilev-alex** [comment]
hey @krystian50! Thanks for the PR. We'll need some time to review

File: examples/auth0/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "auth0",
+  "version": "0.0.1",
+  "author": {
+    "name": "Krystian Fras",
+    "email": "krystian@fras.me"
+  },
+  "private": false,
+  "license": "(MIT OR Apache-2.0)",
```
Comment (vasilev-alex) [inline examples/auth0/package.json:9]: Apache-2.0

File: examples/auth0/dashboard-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+  // show loading indicator while loading
+  if (loading || !isAuthenticated || !cubejsApi) {
+    return <span>Loading</span>;
+  }
+
+  return (
+    <CubeProvider cubejsApi={cubejsApi}>
+      <AppLayout>
+        { JSON.stringify(user) }
```
[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/App.js:71]: Yep, just to print user data on page :)

File: examples/auth0/dashboard-app/src/react-auth0-spa.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        user,
+        loading,
+        popupOpen,
+        loginWithPopup,
+        handleRedirectCallback,
+        getIdTokenClaims: (...p) => auth0Client.getIdTokenClaims(...p),
+        loginWithRedirect: (...p) => auth0Client.loginWithRedirect(...p),
+        getTokenSilently: (...p) => auth0Client.getTokenSilently(...p),
+        getTokenWithPopup: (...p) => auth0Client.getTokenWithPopup(...p),
+        logout: (...p) => auth0Client.logout(...p),
```
[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/react-auth0-spa.js:84]: This file is copy-paste from the official auth0 tutorial. I didn't want to change anything here. I just checked and currently, they created `@auth0/auth0-react` package which basically exposes `useAuth0` hook and works almost the same.

File: examples/auth0/dashboard-app/src/App.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  // Init CubeJS instance with access_token
+  useEffect(() => {
+    if (!cubejsApi && !loading && isAuthenticated) {
+      initCubejs();
+    }
+  }, [cubejsApi, initCubejs, isAuthenticated, loading]);
+
+  // show loading indicator while loading
+  if (loading || !isAuthenticated || !cubejsApi) {
```
[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/App.js:64]: It's automatically redirecting users to the login page. This flow is 'logged in only" - no guests allowed.

File: examples/auth0/dashboard-app/src/init-cubejs-api.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import cubejs, { HttpTransport } from '@cubejs-client/core';
+ 
+const API_URL = 'http://localhost:4000';
+ 
+export default async (accessToken) => {
+ return cubejs({
+   transport: new HttpTransport({
```
[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/init-cubejs-api.js:7]: I tried this:
```
export default (accessToken) => {
 return cubejs({
  apiUrl: `${API_URL}/cubejs-api/v1`,
  headers: { 
    authorization: `Bearer ${accessToken}`
  },
 });
};
```

and the request has such params: 
image

[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/init-cubejs-api.js:7]: My bad. It should be capitalized (`Authorization`)

File: examples/auth0/dashboard-app/src/react-auth0-spa.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/react-auth0-spa.js:84]: Should I upgrade it? I'm not quite sure. What do you think @vasilev-alex ?

[the remark this answers is not in the document]
Comment (vasilev-alex) [reply examples/auth0/dashboard-app/src/react-auth0-spa.js:84]: @krystian50 yes, I would do that

[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/react-auth0-spa.js:84]: Done

File: examples/auth0/dashboard-app/src/App.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  const initCubejs = useCallback(async () => {
+    const accessToken = await getTokenSilently();
+
+    setCubejsApi(await initCubejsApi(accessToken));
+  }, [getTokenSilently]);
+
+  // Init CubeJS instance with access_token
+  useEffect(() => {
+    if (!cubejsApi && !loading && isAuthenticated) {
+      initCubejs();
```
[the remark this answers is not in the document]
Comment (krystian50) [reply examples/auth0/dashboard-app/src/App.js:56]: The flow is not changed. I used new auth0 library

File: examples/auth0/package.json (hunk 1 above)
Comment (vasilev-alex) [reply examples/auth0/package.json]: Please replace this one, and we're ready to go

**Review by vasilev-alex — APPROVED** [review]

**Comment by krystian50** [comment]
@vasilev-alex do you know why the build failed? Looks like it was something unrelated to my changes. Could you rerun it?

File: examples/auth0/.env.example [hunk 7]
```diff
@@ -0,0 +1,10 @@
+CUBEJS_DB_HOST=
+CUBEJS_DB_PORT=
+CUBEJS_DB_NAME=
+CUBEJS_DB_USER=
+CUBEJS_DB_PASS=
+CUBEJS_WEB_SOCKETS=true
+CUBEJS_DB_TYPE=
+CUBEJS_API_SECRET=
```
Comment (kunalfermat) [inline examples/auth0/.env.example:8]: Can someone explain where is this used?

**Pull request merged**